test/main.sh

```diff
@@ -94,9 +94,22 @@ cleanup() {
 
   if [ "${TEST_RESULT}" != "success" ]; then
     # dmesg may contain oops, IO errors, crashes, etc
-    echo "::group::dmesg logs"
+    # If there's a kernel stack trace, don't generate a collapsible group
+
+    expandDmesg=no
+    if journalctl --quiet --no-hostname --no-pager --boot=0 --lines=100 --dmesg --grep "kernel: Call Trace:" > /dev/null; then
+      expandDmesg=yes
+    fi
+
+    if [ "${expandDmesg}" == "no" ]; then
```

Comment:
For historical reasons, we've used the POSIX syntax of a single `=` inside `[]`. This matters less now that we use `bash` everywhere but then it seems that double `=` inside single `[` and `]` is non-standard (https://stackoverflow.com/questions/2237080/how-to-compare-strings-in-bash/2237103#2237103)